restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/entities/Geolocation.java

```diff
@@ -0,0 +1,676 @@
+                    ageOfLocationInfo, deviceLatitude, deviceLongitude, accuracy, physicalAddress, internetAddress,
+                    formattedAddress, locationTimestamp, eventGeofenceLatitude, eventGeofenceLongitude, radius,
+                    geolocationPositioningType, lastGeolocationResponse, cause, apiVersion, uri);
+        }
+
+        public void setSid(Sid sid) {
+            this.sid = sid;
+        }
+
+        public void setDateUpdated(DateTime dateUpdated) {
+            this.dateUpdated = dateUpdated;
+        }
+
+        public void setAccountSid(Sid accountSid) {
+            this.accountSid = accountSid;
+        }
+
+        public void setSource(String source) {
+            this.source = source;
+        }
+
+        public void setDeviceIdentifier(String deviceIdentifier) {
+            this.deviceIdentifier = deviceIdentifier;
+        }
+
+        public void setGeolocationType(GeolocationType geolocationType) {
+            this.geolocationType = geolocationType;
+        }
+
+        public void setResponseStatus(String responseStatus) {
+            this.responseStatus = responseStatus;
+        }
+
+        public void setCellId(String cellId) {
+            this.cellId = cellId;
+        }
+
+        public void setLocationAreaCode(String locationAreaCode) {
+            this.locationAreaCode = locationAreaCode;
+        }
+
+        public void setMobileCountryCode(Integer mobileCountryCode) {
+            this.mobileCountryCode = mobileCountryCode;
+        }
+
+        public void setMobileNetworkCode(String mobileNetworkCode) {
+            this.mobileNetworkCode = mobileNetworkCode;
+        }
+
+        public void setNetworkEntityAddress(Long networkEntityAddress) {
+            this.networkEntityAddress = networkEntityAddress;
+        }
+
+        public void setAgeOfLocationInfo(Integer ageOfLocationInfo) {
+            this.ageOfLocationInfo = ageOfLocationInfo;
+        }
+
+        public void setDeviceLatitude(String devLatitude) {
+            this.deviceLatitude = devLatitude;
+        }
+
+        public void setDeviceLongitude(String devLongitude) {
+            this.deviceLongitude = devLongitude;
+        }
+
+        public void setAccuracy(Long accuracy) {
+            this.accuracy = accuracy;
+        }
+
+        public void setPhysicalAddress(String physicalAddress) {
+            this.physicalAddress = physicalAddress;
+        }
+
+        public void setInternetAddress(String internetAddress) {
+            this.internetAddress = internetAddress;
+        }
+
+        public void setFormattedAddress(String formattedAddress) {
+            this.formattedAddress = formattedAddress;
+        }
+
+        public void setLocationTimestamp(DateTime locationTimestamp) {
+            try {
+                this.locationTimestamp = locationTimestamp;
+            } catch (Exception exception) {
```

Comment:
If a non DateTime value for this attribute is received in a POST or PUT (or from a third party like a GMLC), an exception is thrown.